### PR TITLE
Fix clippy never_loop

### PR DESCRIPTION
**Pull request by hecatia-elegua**
just a small fix / refactor since clippy errored out
also passes cargo fmt now

edit: I just realized clippy has more complaints, but I'll just keep the PR this way and go on with my actual problem

### CODE REVIEW

File: crates/hir-def/src/body/lower.rs [hunk 1]
```diff
@@ -1083,24 +1083,25 @@ impl ExprCollector<'_> {
...
                     }
                     let expr = Expr::Literal(hir_lit);
                     let expr_ptr = AstPtr::new(&ast::Expr::Literal(ast_lit));
                     let expr_id = self.alloc_expr(expr, expr_ptr);
                     Pat::Lit(expr_id)
-                } else {
-                    Pat::Missing
                 };
-            },
+                inner()
```
Comment (lnicola) [inline crates/hir-def/src/body/lower.rs:1103]: I take it `rustfmt` does the same thing? We should probably keep the comment.

**Review by Veykril — COMMENTED** [review]
Instead of turning this into a closure we can just remove the `loop` keyword and make use of the now stable label_break_value feature instead

**Comment by HKalbasi** [comment]
It hits https://github.com/rust-lang/rustfmt/issues/5676 but we can `#[rustfmt::skip]` that block, or just do nothing until that issue becomes resolved.

**Comment by hecatia-elegua** [comment]
The problem was that clippy errors on this. Also it's kinda more readable, though if the closure is less performant or sth I understand

**Comment by Veykril** [comment]
Using `label_break_value` seems more readable to me than a closure that is called immedaitely, closures usually imply being lazy to me when used in such a way (or emulating a try block).

**Comment by HKalbasi** [comment]
@hecatia-elegua clippy is not considered a problem in this repository, but for maximizing readability I think you can make it a standalone function or method which is more readable than a labeled block and closure.

**Pull request closed without merging**